packages/misc/src/lib/helper/session-sigs-validator.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  AuthSig,
+  Capability,
+  ParsedSessionMessage,
+  ParsedSignedMessage,
+  SessionSigsMap,
+} from '@lit-protocol/types';
+
+interface ValidationResult {
+  isValid: boolean;
+  errors: string[];
+}
+
+// Function to parse a signedMessage string into an object
+function parseSignedMessage(signedMessage: string): ParsedSignedMessage {
+  const lines = signedMessage.split('\n');
+  const parsedData: ParsedSignedMessage = {};
+  let currentKey: string | null = null as string | null;
+  let currentValue = '';
+
+  lines.forEach((line) => {
+    // Match lines with 'Key: Value' pattern
+    const keyValueMatch = line.match(/^([^:]+):\s*(.*)$/);
+
+    if (keyValueMatch) {
+      // Save the previous key-value pair
+      if (currentKey !== null) {
+        parsedData[currentKey.trim()] = currentValue.trim();
+      }
+
+      // Start a new key-value pair
+      currentKey = keyValueMatch[1];
+      currentValue = keyValueMatch[2];
+    } else if (line.startsWith('- ')) {
+      // Handle list items
+      const item = line.substring(2).trim();
+      if (!parsedData[currentKey!]) {
+        parsedData[currentKey!] = [];
+      }
+      (parsedData[currentKey!] as string[]).push(item);
+    } else if (line.trim() === '') {
+      // Skip empty lines
+    } else {
+      // Continuation of the current value
+      currentValue += '\n' + line;
```

Comment:
Why do we add `\n` if we find that there is more data in the current line item?

Reply:
yea just tryna keep the line breaks intact for the multi-line values, test with and without it won't affect the result.

Reply:
Ah, forEach is not in fact async. so actually not an issue in the first place.